source/MRMesh/MRPointsComponents.cpp

```diff
@@ -0,0 +1,182 @@
+#include "MRPointsComponents.h"
+#include "MRPointCloud.h"
+#include "MRBitSet.h"
+#include "MRTimer.h"
+#include "MRPointsInBall.h"
+#include "MRProgressCallback.h"
+#include "MRPch/MRTBB.h"
+#include "MRBitSetParallelFor.h"
+
+namespace MR
+{
+
+/// extended version BitSetParallelForAll with information about range
+template <typename BS, typename F>
+bool BitSetParallelForAllEx( const BS& bs, F f, ProgressCallback progressCb, size_t reportProgressEveryBit = 1024 )
```

Comment:
Too many questions about it:
1. What is "information about range"?
2. Why it is not near the other `BitSetParallelForAll` ones?
3. When to select one and then the other?